### PR TITLE
SK-1713 implement client initialisation and credentials prioritisation

### PR DESCRIPTION
**Pull request by skyflow-bharti**

### CODE REVIEW

**Review by neeraja-skyflow — APPROVED** [review]

**Comment by bhavpreet** [comment]
Will moving the existing lib files to `v1` not impact existing customers?

File: v2/client/client_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
+package client
```
Comment (bhavpreet) [inline v2/client/client_test.go:1]: rename `package client` to `package client_test`

File: v2/client/vault_service_test.go [hunk 2]
```diff
@@ -0,0 +1,257 @@
+package client
```
Comment (bhavpreet) [inline v2/client/vault_service_test.go:1]: Rename `package client` to `package client_test`

File: v2/serviceaccount/token.go [hunk 3]
```diff
@@ -0,0 +1 @@
+package serviceaccount
```
Comment (bhavpreet) [inline v2/serviceaccount/token.go:1]: Q: how is this used?

File: v2/utils/logger/logger.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+//	case DEBUG:
+//		log.SetLevel(logrus.DebugLevel)
+//	case WARN:
+//		log.SetLevel(logrus.WarnLevel)
+//	case ERROR:
+//		log.SetLevel(logrus.ErrorLevel)
+//	case OFF:
+//		log.SetOutput(ioutil.Discard)
+//	default:
+//		log.SetLevel(logrus.ErrorLevel)
```
Comment (bhavpreet) [inline v2/utils/logger/logger.go:31]: Can remove the commented portion.

File: v2/serviceaccount/token.go (hunk 3 above)
Comment (skyflow-bharti) [reply v2/serviceaccount/token.go:1]: this will be used for bearer token generation functions, currently working on this

Comment (bhavpreet) [reply v2/serviceaccount/token.go:1]: add a comment in file.

**Comment by bhavpreet** [comment]
Add a common `test`s to test and use the go library

File: v2/serviceaccount/token.go (hunk 3 above)
Comment (skyflow-bharti) [reply v2/serviceaccount/token.go]: Added comment

File: v2/utils/logger/logger.go (hunk 4 above)
Comment (skyflow-bharti) [reply v2/utils/logger/logger.go]: Removed the commented code

**Pull request closed without merging**